middleware/middleware_session.go

```diff
@@ -0,0 +1,129 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/0xsequence/quotacontrol/proto"
+)
+
+// Session middleware that detects the session type and sets the account or service on the context.
+func Session() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return session{Next: next}
```

Comment:
I don't follow why instantiating session struct is necessary

why not write the logic directly in this MW?

Reply:
I created a struct for each middleware. Just for consistency.

Reply:
I noticed -- but I don't think it's necessary :)
You can create local variables in the first function returning the MW function :)

```
func Session() func(http.Handler) http.Handler {
	// create vars here
	return func(next http.Handler) http.Handler {
		// MW
	}
}
```

Anyway, not a blocker ;)